src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -493,11 +498,9 @@ private boolean perform(Run src, Run<?,?> dst, String expandedFilter, String exp
     public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
 
         public FormValidation doCheckProjectName(
-                @AncestorInPath AbstractProject anc, @QueryParameter String value) {
-            // Require CONFIGURE permission on this project
-            if (!anc.hasPermission(Item.CONFIGURE)) return FormValidation.ok();
```

Comment:
> This permission check applies to current job configuration

You misunderstand completely. The intention is to avoid doing potentially sensitive computations unless the user is actually configuring _this_ (downstream) job. Otherwise someone can fake up a `…/checkProjectName` request in cURL and get some information they should not have access to based on job permissions.

As @ikedam says, just replace `AbstractProject` with `Job`.